src/Kunstmaan/UserManagementBundle/Controller/UsersController.php

```diff
@@ -171,51 +165,84 @@ public function editAction(Request $request, $id)
-            } else {
-                $form->handleRequest($request);
+            $response = $this->handleEditForm($request, $form, $user, $tabPane);
+            if ($response instanceof RedirectResponse) {
+                return $response;
             }
+        }
+        $params = ['form' => $form->createView(), 'user' => $user];
+        $params = $tabPane ? array_merge($params, ['tabPane' => $tabPane]) : $params;
+        return $params;
+    }
 
-            if ($form->isSubmitted() && $form->isValid()) {
-                /* @var UserManager $userManager */
-                $userManager = $this->container->get('fos_user.user_manager');
-                $userManager->updateUser($user, true);
+    /**
+     * @param Request $request
+     * @param Form $form
+     * @param User $user
+     * @param TabPane|null $tabPane
+     * @return null|RedirectResponse
+     */
+    private function handleEditForm(Request $request, Form $form, User $user, TabPane $tabPane = null)
+    {
+        if ($tabPane instanceof TabPane) {
+            $tabPane->bindRequest($request);
+            $form = $tabPane->getForm();
+        } else {
+            $form->handleRequest($request);
+        }
 
-                $this->addFlash(
-                    FlashTypes::SUCCESS,
-                    $this->get('translator')->trans('kuma_user.users.edit.flash.success.%username%', [
-                        '%username%' => $user->getUsername()
-                    ])
-                );
+        if ($form->isSubmitted() && $form->isValid()) {
+            /* @var UserManager $userManager */
+            $userManager = $this->container->get('fos_user.user_manager');
+            $userManager->updateUser($user, true);
 
-                return new RedirectResponse(
-                    $this->generateUrl(
-                        'KunstmaanUserManagementBundle_settings_users_edit',
-                        array('id' => $id)
-                    )
-                );
-            }
-        }
+            $this->addFlash(
+                FlashTypes::SUCCESS,
+                $this->get('translator')->trans('kuma_user.users.edit.flash.success.%username%', [
+                    '%username%' => $user->getUsername()
+                ])
+            );
 
-        $params = array(
-            'form' => $form->createView(),
-            'user' => $user,
-        );
+            $url = $this->generateUrl('KunstmaanUserManagementBundle_settings_users_edit', ['id' => $user->getId()]);
+            return new RedirectResponse($url);
+        }
+        return null;
+    }
 
-        if ($tabPane) {
-            $params = array_merge($params, array('tabPane' => $tabPane));
+    /**
+     * @return string
+     */
+    private function getRequiredRole($id)
+    {
+        // The logged in user should be able to change his own password/username/email and not for other users
+        if ($id == $this->get('security.token_storage')->getToken()->getUser()->getId()) {
```

Comment:
Can be simplified to return 'ROLE_ADMIN' and no else statement but return 'ROLE_SUPER_ADMIN'